cluster-autoscaler/cloudprovider/gce/os_reserved.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+// OsReservedCalculator calculates the OS reserved values.
+type OsReservedCalculator interface {
+	// CalculateKernelReserved computes how much memory OS kernel will reserve.
+	// NodeVersion parameter may be optional if it is not used in calculation.
```

Comment:
As a user of the interface I don't know if the field is used in calculation or not. What I had in mind was along the lines of 'NodeVersion parameter is optional. If left empty a result calculated using default node version will be returned', ie. setting requirements for how any implementation is expected to handle empty value.